### PR TITLE
Refactor progress bars set new defaults

### PR DESCRIPTION
**Pull request by AntreasAntoniou**
Fixed lack of neatness introduced by the rich fancy terminal interface. Also updated some default hyperparameters to ones easier to work with (that work better out of the box). 

Also, fixed a couple of bugs here and there.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -98,27 +97,14 @@ def get_base_argument_parser():
...
-    epoch,
-    data_loader,
-    model,
-    metric_tracker,
-    progress_panel,
-    progress_tracker,
-    overall_progress,
-    overall_task,
-):
+def train(epoch, data_loader, model, metric_tracker, progress_reporter):
```
Comment (jack-willturner) [inline train.py:100]: this function is sooo much nicer now

File: train.py [hunk 2]
```diff
@@ -370,82 +329,39 @@ def eval(
...
-    progress_table.add_row(
-        Panel.fit(epoch_progress, title="[b]Jobs", border_style="red", padding=(1, 2))
-    )
-    progress_table.add_row(metric_tracker_train.per_epoch_table)
-    progress_table.add_row(metric_tracker_val.per_epoch_table)
 
     train_iterations = 0
 
-    with Live(progress_table, refresh_per_second=10) as interface_panel:
+    with Live(
```
Comment (jack-willturner) [inline train.py:346]: This is good for now but it would be better in future if:
- we could provide a more generic `with` handler (not specific to rich) 
- we could wrap rich in a builder 
- the user could then choose the type of metric tracker (e.g. plain printing, rich, tqdm) from the train script

Comment (learning-luke) [reply train.py:346]: I agree. This would be really cool.

File: train.py [hunk 3]
```diff
@@ -456,58 +372,60 @@ def eval(
...
 
         if args.test:
             test_progress_dict = {
                 **dict(description="[green]Training", total=len(test_set_loader)),
                 **metric_tracker_test.get_current_iteration_metric_text_column_fields(),
             }
 
-            test_epoch_progress = epoch_progress.add_task(**test_progress_dict)
+            test_epoch_progress = progress_reporter.epoch_progress.add_task(
+                **test_progress_dict
```
Comment (jack-willturner) [inline train.py:409]: why do you need this here and not for the main training loop?

File: utils/metric_tracking.py [hunk 4]
```diff
@@ -9,12 +10,13 @@
 from rich.progress import TextColumn
```
Comment (jack-willturner) [inline utils/metric_tracking.py:10]: can these rich imports now be removed?

**Review by jack-willturner — CHANGES_REQUESTED** [review]
Remove the unused rich imports in metric_tracking.py

File: train.py (hunk 2 above)
Comment (AntreasAntoniou) [reply train.py:346]: I'll open an issue for this then! :)

File: train.py (hunk 3 above)
Comment (AntreasAntoniou) [reply train.py:409]: It avoids having 'testing' in the terminal interface before it is necessary. :smile:

**Comment by AntreasAntoniou** [comment]
Done @jack-willturner #0ada87b

**Review by jack-willturner — APPROVED** [review]
